### PR TITLE
Update defaults and logic to match old FE

### PR DESCRIPTION
**Pull request by amyeroberts**
# What does this PR do?

Updates defaults and logic in image processors to match the previous feature extractors. Fixes some broken inference tests. 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: tests/models/perceiver/test_modeling_perceiver.py [hunk 1]
```diff
@@ -903,7 +903,7 @@ def test_inference_image_classification(self):
         expected_shape = torch.Size((1, model.config.num_labels))
         self.assertEqual(logits.shape, expected_shape)
 
-        expected_slice = torch.tensor([-1.1653, -0.1993, -0.7521], device=torch_device)
```
Comment (amyeroberts) [inline tests/models/perceiver/test_modeling_perceiver.py:906]: In this case I updated the test values. I considered this OK as other tests that were previously failing now pass with and the difference was 0.0001 across all elements. 

I wasn't able to track down exactly where this is introduced. LMK if you think this is an issue and I can do some more digging.

**Review by sgugger — APPROVED** [review]
Thnaks for all the fixes!

**Review by ydshieh — APPROVED** [review]
Thanks for the fix. Let me run it too, will report soon.

File: tests/models/perceiver/test_modeling_perceiver.py (hunk 1 above)
Comment (ydshieh) [reply tests/models/perceiver/test_modeling_perceiver.py:906]: With a VM as close as to the runner, I get 

```python
tensor([-1.1652, -0.1993, -0.7521], device='cuda:0')
```

and with numpy, it is
```
array([-1.1652362 , -0.19931625, -0.75210035], dtype=float32)
```

Let me check within docker ...

Comment (ydshieh) [reply tests/models/perceiver/test_modeling_perceiver.py:906]: On the runner within docker

```bash
(Pdb) logits[0, :3]
tensor([-1.1652, -0.1992, -0.7520], device='cuda:0')

(Pdb) logits[0, :3].to("cpu").numpy()
array([-1.1652446 , -0.19924526, -0.7520099 ], dtype=float32)
``` 

maybe let's use this value and see if we get lucky in the next CI?

Comment (amyeroberts) [reply tests/models/perceiver/test_modeling_perceiver.py:906]: OK, sounds good. Just to confirm - I'll update the values to the ones when running within docker: 
`[-1.1652446 , -0.19924526, -0.7520099 ]`

Comment (ydshieh) [reply tests/models/perceiver/test_modeling_perceiver.py:906]: Yes

Comment (amyeroberts) [reply tests/models/perceiver/test_modeling_perceiver.py:906]: I've updated them and pushed. Once all the tests have passed, I'll merge and keep an eye out for the slow run CI tests if it needs to be updated again. 

Thanks for your help in looking into these numbers.

**Pull request merged**